tests/gemm_test_all.cpp

```diff
@@ -30,25 +30,65 @@
+          std::cout << "A Transposed: "<< !transA << ", B Transposed: " << !transB << ", M: " << m  << " K: "<< k << ", N: " << n << std::endl;
+          run_gemm_tests<!transA, !transB, float>(seed, m, k, n, 1);
+          std::cout << std::boolalpha;
+          std::cout << "A Transposed: "<< !transA << ", B Transposed: " << transB << ", M: " << m  << " K: "<< k << ", N: " << n << std::endl;
+          run_gemm_tests<!transA,  transB, float>(seed, m, k, n, 1);
+          std::cout << std::boolalpha;
+          std::cout << "A Transposed: "<< transA << ", B Transposed: " << !transB << ", M: " << m  << " K: "<< k << ", N: " << n << std::endl;
+          run_gemm_tests< transA, !transB, float>(seed, m, k, n, 1);
+          std::cout << std::boolalpha;
+          std::cout << "A Transposed: "<< transA << ", B Transposed: " << transB << ", M: " << m  << " K: "<< k << ", N: " << n << std::endl;
+          run_gemm_tests< transA,  transB, float>(seed, m, k, n, 1);
+          std::cout << std::boolalpha;
+          std::cout << "A Transposed: "<< !transA << ", B Transposed: " << !transB << ", M: " << m + 1 << " K: "<< k + 1 << ", N: " << n + 1 << std::endl;
+          run_gemm_tests<!transA, !transB, float>(seed, m + 1, k + 1, n + 1, 1);
+          std::cout << std::boolalpha;
+          std::cout << "A Transposed: "<< !transA << ", B Transposed: " << transB << ", M: " << m + 1 << " K: "<< k + 1 << ", N: " << n + 1 << std::endl;
+          run_gemm_tests<!transA,  transB, float>(seed, m + 1, k + 1, n + 1, 1);
+          std::cout << std::boolalpha;
+          std::cout << "A Transposed: "<< transA << ", B Transposed: " << !transB << ", M: " << m + 1 << " K: "<< k + 1 << ", N: " << n + 1 << std::endl;
+          run_gemm_tests< transA, !transB, float>(seed, m + 1, k + 1, n + 1, 1);
+          std::cout << std::boolalpha;
+          std::cout << "A Transposed: "<< transA << ", B Transposed: " << transB << ", M: " << m + 1 << " K: "<< k + 1 << ", N: " << n + 1 << std::endl;
+          run_gemm_tests< transA,  transB, float>(seed, m + 1, k + 1, n + 1, 1);
+  } else {
+    const int m = std::atoi(argv[1]);
+    const int k = std::atoi(argv[2]);
+    const int n = std::atoi(argv[3]);
+    // rep is the warm up period
+    const int rep = std::atoi(argv[4]);
 
-  const int seed = 42;
-  const int m = std::atoi(argv[1]);
-  const int k = std::atoi(argv[2]);
-  const int n = std::atoi(argv[3]);
-  const int rep = std::atoi(argv[4]);
+    run_gemm_tests<!transA, !transB, float>(seed, m, k, n, rep);
+    run_gemm_tests<!transA,  transB, float>(seed, m, k, n, rep);
+    run_gemm_tests< transA, !transB, float>(seed, m, k, n, rep);
+    run_gemm_tests< transA,  transB, float>(seed, m, k, n, rep);
+  }
 
-  run_gemm_tests<!transA, !transB, float>(seed, m, k, n, rep);
-  run_gemm_tests<!transA,  transB, float>(seed, m, k, n, rep);
-  run_gemm_tests< transA, !transB, float>(seed, m, k, n, rep);
-  run_gemm_tests< transA,  transB, float>(seed, m, k, n, rep);
 
```

Comment:
can you make this return something different from zero if the test fails? it eases integration with testing tools (like ctest)